scripts/generate-keyfile.ts

```diff
@@ -0,0 +1,24 @@
+#! /usr/bin/env ts-node
+
+import ora = require('ora');
+import fs = require('fs');
+import path = require('path');
+import { resolve } from 'path';
+import { config } from 'dotenv';
+
+const { promisify } = require('util');
+const writeFile = promisify(fs.writeFile);
+const ROOT_DIR = path.join(__dirname, '..');
+
+config({ path: resolve(__dirname, '../.env') });
+
+(async () => {
+  if (process.env.SEGMENT_KEY) {
+    const ui = ora('Generate constants keyfile').start();
+    await writeFile(
+      `${ROOT_DIR}/constants.json`,
+      JSON.stringify({ segmentKey: process.env.SEGMENT_KEY }, null, 2)
```

Comment:
To avoid storing segment key on git, we keep il locally in .env file, that is not being pushed to git, and during the compilation process, we generate constants.json file with this key injected. The constants.json file is also added to `.gitignore`, so the key exists only on the local machine.

To publish vscode extension or to build a package locally we use `vsce` command-line tool, sort of `npm publish` for vscode, that require the compilation step as well, therefore the build will also include constants.json file.

When we run tests on Azure, we don't have access to .env file that is stored on our computer, therefore to get constants.json generated there, I use Azure Pipeline variable that I pass to npm test script in azure-pipelines.yml.

I think later when we will have automatic publishing using CI we can get rid of .env file and pass a the same azure variable to deploy script. Since we don't publish to the marketplace yet, and I can't test how this key injection will work, I have this intermediate solution.

Reply:
Or we can keep .env, in any case, if we want to continue building packages locally, what also can be useful.